src/services/auth/types.ts

```diff
@@ -434,3 +439,61 @@ export interface AuthOauth1Service {
    */
   removeToken(tokenId: string): Promise<AffectedRecords>;
 }
+
+export interface OidcService {
+  /**
+   * Create an OpenId Connect Provider
+   * Permission | Scope | Effect
+   * - | - | -
+   * `CREATE_OIDC_PROVIDER` | `global` | **Required** for this endpoint
+   */
+  createProvider(
+    requestBody: OidcProviderCreation
+  ): Promise<OidcProviderResponse>;
+
+  /**
+   * Get a list of OpenId Connect Providers
+   * @param rql Add filters to the requested list.
+   * Permission | Scope | Effect
+   * - | - | -
+   * `VIEW_OIDC_PROVIDERS` | `global` | **Required** for this endpoint
+   */
+  getProviders(
+    options?: OptionsWithRql
+  ): Promise<PagedResult<OidcProviderResponse>>;
+
+  /**
+   * Update an OpenId Connect Provider
+   * Permission | Scope | Effect
+   * - | - | -
+   * `UPDATE_OIDC_PROVIDER` | `global` | **Required** for this endpoint
+   */
+  updateProvider(
+    providerId: string,
+    requestBody: Partial<OidcProviderUpdate>
```

Comment:
(minor) The OidcProviderUpdate is already a partial, so this is technically not need 😄